.github/workflows/build-vscode-extension.yml

```diff
@@ -98,3 +98,53 @@ jobs:
         with:
           name: ${{ matrix.builds.name }}
           path: vscode/${{ matrix.builds.path }}/*.${{ matrix.builds.item }}
+
+  generate_binaries:
+      needs: [checksecret]
+      strategy:
+        matrix:
+          binaries:
+            - architecture: win-x86
+              os: windows-latest
+            - architecture: win-x64
+              os: windows-latest
+            - architecture: linux-x64
+              os: ubuntu-latest
+            - architecture: linux-arm64
+              os: ubuntu-latest
+            - architecture: osx-x64
+              os: macOS-latest
+            - architecture: osx-arm64
+              os: macOS-latest
+      runs-on: ${{ matrix.binaries.os }}
+      steps:
+        - name: Checkout code
+          uses: actions/checkout@v4
+  
+        - name: Set up Node.js
+          uses: actions/setup-node@v4
+          with:
+            node-version: 20.x
+  
+        - name: Use .NET 6
+          uses: actions/setup-dotnet@v3
+          with:
+            dotnet-version: '6.x'
+  
+        - name: Use .NET 9
+          uses: actions/setup-dotnet@v3
+          with:
+            dotnet-version: '9.x'
+
+        - name: Publish kiota for ${{ matrix.binaries.architecture }}
+          run: dotnet publish src/kiota/kiota.csproj -c Release --runtime ${{ matrix.binaries.architecture }} -p:PublishSingleFile=true --self-contained --output binaries/${{ matrix.binaries.architecture }} -f net9.0
+          shell: pwsh
+
+        - name: Upload artifact
```

Comment:
Given that this will consume aprox 200b of storage in th runs, we could specify the `retention-days:` propety and limit it to 3 days?

Reply:
I see.... updated